### ISSUE TITLE
Cannot find symbol parseOverScrollMode

### ISSUE THREAD

**Issue by Doko-Demo-Doa** [cesardeazevedo/react-native-bottom-sheet-behavior#19]
`/Users/doko/Documents/ReactNative/Zoti/node_modules/react-native-bottom-sheet-behavior/android/src/main/java/com/bottomsheetbehavior/ReactNestedScrollViewManager.java:118: error: cannot find symbol
        view.setOverScrollMode(ReactScrollViewHelper.parseOverScrollMode(value));
                                                    ^
  symbol:   method parseOverScrollMode(String)
  location: class ReactScrollViewHelper
Note: /Users/doko/Documents/ReactNative/Zoti/node_modules/react-native-bottom-sheet-behavior/android/src/main/java/com/bottomsheetbehavior/ResourceDrawableIdHelper.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.`

Didn't have same issues with another libraries. I think the problem lies within Support Library v24 but I  cannot downgrade it (to v23, same as currently shipped with React Native Android) because BottomSheet is only available on v24 and up.

Any workarounds?

**Comment by cesardeazevedo** [comment]
Hi, Thanks for reporting.

Which react-native version you're using?

**Comment by Doko-Demo-Doa** [comment]
Oh sorry I forgot to mention. It is 0.42.

Things are getting more interesting. When I decode the ReactScrollViewHelper.java file (included in ReactNative dependencies downloaded by gradle, there's no method named "parseOverScrollMode":

ReactScrollViewHelper.java

```
package com.facebook.react.views.scroll;

import android.os.SystemClock;
import android.view.View;
import android.view.ViewGroup;

import com.facebook.react.bridge.ReactContext;
import com.facebook.react.uimanager.UIManagerModule;

public class ReactScrollViewHelper {

  public static final long MOMENTUM_DELAY = 20;

  public static void emitScrollEvent(ViewGroup scrollView) {
    emitScrollEvent(scrollView, ScrollEventType.SCROLL);
  }

  public static void emitScrollBeginDragEvent(ViewGroup scrollView) {
    emitScrollEvent(scrollView, ScrollEventType.BEGIN_DRAG);
  }

  public static void emitScrollEndDragEvent(ViewGroup scrollView) {
    emitScrollEvent(scrollView, ScrollEventType.END_DRAG);
  }

  public static void emitScrollMomentumBeginEvent(ViewGroup scrollView) {
    emitScrollEvent(scrollView, ScrollEventType.MOMENTUM_BEGIN);
  }

  public static void emitScrollMomentumEndEvent(ViewGroup scrollView) {
    emitScrollEvent(scrollView, ScrollEventType.MOMENTUM_END);
  }

  private static void emitScrollEvent(ViewGroup scrollView, ScrollEventType scrollEventType) {
    View contentView = scrollView.getChildAt(0);

    if (contentView == null) {
      return;
    }

    ReactContext reactContext = (ReactContext) scrollView.getContext();
    reactContext.getNativeModule(UIManagerModule.class).getEventDispatcher().dispatchEvent(
        ScrollEvent.obtain(
            scrollView.getId(),
            SystemClock.uptimeMillis(),
            scrollEventType,
            scrollView.getScrollX(),
            scrollView.getScrollY(),
            contentView.getWidth(),
            contentView.getHeight(),
            scrollView.getWidth(),
            scrollView.getHeight()));
  }
}
```

But the source code in github does (at the bottom):

https://github.com/facebook/react-native/blob/ee737e7d1c2b60d2226bc9fcab372ce98b5976f1/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollViewHelper.java

I created an issue in RN's repo but I still hope for a workaround without having to recompile whole RN dependency. I'm fairly familiar with Android development but trying to downgrade RN dependency (currently 0.20.1) doesn't help.

TL;DR: This bug isn't really caused by your plugin, but the RN library itself, I just don't get why.

**Comment by cesardeazevedo** [comment]
I am not sure whether it is your case, but did you mentioned `0.20.1`, so are you trying to run through android studio right? it seems that your android studio is getting a very old version from react-native since they [stop updates on the jcenter](https://bintray.com/bintray/jcenter/com.facebook.react:react-native/view) in favor of a local repository, which they stuck at the `0.20.1` version, and for some reason the android studio still getting this reference instead of the `node_modules/react-native/android`, i am not sure why, but i also had this problem, but take a look at `external library` on android studio and see which version is showing, and try to clean the project and refresh the gradle dependencies.

My [example](https://github.com/cesardeazevedo/react-native-bottom-sheet-behavior/tree/master/example) is fully working with react-native 0.42.0, but i didn't tested with android studio though.

Did you tried running with `react-native run-android` directly?
I hope it helps, tell me if you got it.

**Comment by Doko-Demo-Doa** [comment]
Oh yes! I got why.

I'm actually using 0.41. Which does not have mentioned method:

https://github.com/facebook/react-native/blob/0.41-stable/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollViewHelper.java

But 0.42 (and above) does:

https://github.com/facebook/react-native/blob/0.42-stable/ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollViewHelper.java

Problem solved. Actually I only need NestedScrollView component, but too lazy to write my own module so I end up using your plugin instead :P

Thank you very much.

Edit: This is not necesary at all. But since your plugin is just a module, it's not mandatory to include whole project structure into android/ folder. Only these are enough:

src/
build.gradle

And build.gradle file should be only:

```
apply plugin: "com.android.library"

android {
    compileSdkVersion 23
    buildToolsVersion "23.0.1"

    defaultConfig {
        minSdkVersion 16
        targetSdkVersion 23
        versionCode 1
        versionName "1.0"
        ndk {
            abiFilters "armeabi-v7a", "x86"
        }
    }
    lintOptions {
       warning 'InvalidPackage'
    }
}

dependencies {
    compile "com.facebook.react:react-native:+"
    compile "com.android.support:appcompat-v7:24.1.1"
    compile 'com.android.support:design:24.1.1'
}

```

**Comment by cesardeazevedo** [comment]
Yes indeed, i will take a look at the build.gradle in future versions.
